concepts/pivoted/eICU-pivoted-OASIS.sql

```diff
@@ -0,0 +1,503 @@
+    WHEN MAX(respiratoryrate) BETWEEN 31 AND 44 THEN 6
+    WHEN MAX(respiratoryrate) > 44 THEN 9
+    ELSE NULL
+    END AS respiratoryrate_oasis
+
+  FROM `physionet-data.eicu_crd_derived.pivoted_vital`
+  WHERE (chartoffset > 0 AND chartoffset <= 1440) -- consider only first 24h
+    AND respiratoryrate IS NOT NULL
+  GROUP BY pid_RR
+)
+
+-- Temperature 
+, temperature_oasis AS (
+  SELECT patientunitstayid AS pid_temp
+  , CASE
+    WHEN MIN(temperature) < 33.22 THEN 3
+    WHEN MIN(temperature) BETWEEN 33.22 AND 35.93 THEN 4
+    WHEN MAX(temperature) BETWEEN 33.22 AND 35.93 THEN 4
+    WHEN MIN(temperature) BETWEEN 35.94 AND 36.39 THEN 2
+    WHEN MAX(temperature) BETWEEN 36.40 AND 36.88 THEN 0
+    WHEN MAX(temperature) BETWEEN 36.89 AND 39.88 THEN 2
+    WHEN MAX(temperature) >39.88 THEN 6
+    ELSE NULL
+    END AS temperature_oasis
+  FROM `physionet-data.eicu_crd_derived.pivoted_vital`
+  WHERE (chartoffset > 0 AND chartoffset <= 1440) -- consider only first 24h
+    AND temperature IS NOT NULL
+  GROUP BY pid_temp
+)
+
+-- Urine output
+, merged_uo AS (
+  
+  -- pat table as base for patientunitstayid 
+  SELECT pat.patientunitstayid, COALESCE(pivoted_uo.urineoutput, apache_urine.urine) AS uo_comb -- consider pivoted_uo first, if missing -> apacheapsvar
+  FROM `physionet-data.eicu_crd.patient` AS pat
+  
+  -- Join information from pivoted_uo table
+  LEFT JOIN(
+  SELECT patientunitstayid AS pid_uo, SUM(urineoutput) AS urineoutput
+  FROM `physionet-data.eicu_crd_derived.pivoted_uo` 
+  WHERE (chartoffset > 0 AND chartoffset <= 1440) -- consider only first 24h
+  AND urineoutput > 0 AND urineoutput IS NOT NULL -- ignore biologically implausible values <0
+  GROUP BY pid_uo
+  ) AS pivoted_uo
+  ON pivoted_uo.pid_uo = pat.patientunitstayid
+
+  -- Join information from apacheapsvar table
+  LEFT JOIN(
+  SELECT patientunitstayid AS pid_auo, urine
+  FROM `physionet-data.eicu_crd.apacheapsvar` 
+  WHERE urine > 0 AND urine IS NOT NULL -- ignore biologically implausible values <0
+  ) AS apache_urine
+  ON apache_urine.pid_auo = pat.patientunitstayid
+
+)
+
+-- Call merged_uo table for score computation
+, urineoutput_oasis AS (
+  SELECT merged_uo.patientunitstayid AS pid_urine, merged_uo.uo_comb
+  , CASE
+    WHEN uo_comb <671 THEN 10
+    WHEN uo_comb BETWEEN 671 AND 1426.99 THEN 5
+    WHEN uo_comb BETWEEN 1427 AND 2543.99 THEN 1
+    WHEN uo_comb BETWEEN 2544 AND 6896 THEN 0
+    WHEN uo_comb >6896 THEN 8
+    ELSE NULL
+    END AS urineoutput_oasis
+  FROM merged_uo
+)
+
+-- Ventiliation -> Note: This information is stored in 5 tables
+-- Create unified vent_table first
+, merged_vent AS (
+
+    -- 1: use patient table as base 
+    SELECT pat.patientunitstayid, vent_1, vent_2, vent_3, vent_4
+    FROM `physionet-data.eicu_crd.patient` AS pat
+
+    -- 2: ventilation_events table
+      LEFT JOIN(
+        SELECT patientunitstayid,
+        CASE WHEN ( MAX(event) = "mechvent start" OR MAX(event) = "mechvent end")THEN 1
```

Comment:
`MAX()` should be wrapped around the case statement, i.e. `MAX(CASE WHEN event = ... ELSE NULL END) AS vent_1`.

`MAX(event) = "mechvent start"` will fail if there is another non-mech vent value. For example, if there are three rows `mechvent start`, `mechvent end`, and `zzz`, `MAX(event) = "zzz"`, therefore this case statement gives `NULL` when it should be 1.